manta-crypto/src/arkworks/constraint/mod.rs

```diff
@@ -33,335 +31,26 @@ use manta_crypto::{
             },
         },
     },
-    constraint::{
-        measure::{Count, Measure},
-        Input, ProofSystem,
-    },
+    constraint::measure::{Count, Measure},
     eclair::{
         self,
         alloc::{
             mode::{self, Public, Secret},
             Constant, Variable,
         },
-        bool::{Assert, Bool, ConditionalSelect, ConditionalSwap},
+        bool::{Assert, ConditionalSelect, ConditionalSwap},
         num::{AssertWithinBitRange, Zero},
-        ops::{Add, BitAnd, BitOr},
+        ops::Add,
         Has, NonNative,
     },
-    rand::{RngCore, Sample},
-};
-use manta_util::{
-    byte_count,
-    codec::{Decode, DecodeError, Encode, Read, Write},
-    SizeLimit,
 };
 
-#[cfg(feature = "serde")]
-use manta_util::serde::{Deserialize, Serialize, Serializer};
-
-pub use manta_crypto::arkworks::{
+pub use crate::arkworks::{
     r1cs_std::{bits::boolean::Boolean, fields::fp::FpVar},
     relations::r1cs::SynthesisError,
 };
```

Comment:
why are we re-exporting our own packages from here instead of simply importing them from the corresponding files when necessary?

Reply:
Because these specific ones are primitives used in ECLAIR (part of ECLAIR std library). Importing them from the base location is more natural than from their arbitrary location set by `arkworks`. This will be more organized in the future, but you definitely don't want to import `Boolean` from `r1cs_std::bits::boolean::Boolean`.